app/services/dttp/retrieve_users.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Dttp
+  class RetrieveUsers
+    include ServicePattern
+
+    class Error < StandardError; end
+
+    DEFAULT_PATH = "/contacts?$filter=dfe_portaluser eq true"
+    MAX_PAGE_SIZE = 25
```

Comment:
Given https://github.com/DFE-Digital/register-trainee-teachers/blob/2e676a8aacb2d553ad61839b10b5c18821f683f9/app/services/dttp/import_user.rb#L21-L25

I be more in flavour being selective and uniting `user  & provider` as a single unit, as user must have a provider.


`$filter=dfe_portaluser eq true&$select=firstname,lastname,emailaddress1,_parentcustomerid_value,parentcustomerid_account&$expand=parentcustomerid_account($select=dfe_marketingname)`

That will probably eliminate the need for all that pagination calling, as well as the payload issue.

Reply:
If you mean combining them into one model/table I don't think we should do that. I don't think storing them denormalised at this stage is necessary and I also think we probably have use cases for having a local store of distinct providers from DTTP. I can see that would simplify this cacheing by removing the other process for getting the providers but it doesn't feel like the correct way to store them to me.

Reply:
Just been discussing this offline. We should definitely add the select to the query to reduce the response size.

Reply:
Discussing this bit


`$filter=dfe_portaluser eq true&$select=firstname,lastname,emailaddress1,_parentcustomerid_value` 



we should do this bit through its drops payload by 80% and negate need for pagination

Reply:
On production the query unpaginated too 2.7s and then ~750ms on subsequent attempts once the hamsters had woken up. 🐹

Reply:
done https://github.com/DFE-Digital/register-trainee-teachers/pull/751/commits/9745b1f192a1a7369e8b2063e3f1047b05a690b1